Lib/dataclasses.py

```diff
@@ -1399,9 +1402,11 @@ def _add_slots(cls, is_frozen, weakref_slot, defined_fields):
             f.type = ann
 
     # Fix the class reference in the __annotate__ method
-    init_annotate = newcls.__init__.__annotate__
-    if getattr(init_annotate, "__generated_by_dataclasses__", False):
-        _update_func_cell_for__class__(init_annotate, cls, newcls)
+    init = newcls.__init__
+    if hasattr(init, "__annotate__"):
+        init_annotate = init.__annotate__
```

Comment:
```suggestion
    if init_annotate := getattr(init, "__annotate__", None):
```

There's no need to get this attr twice.